teuthology/task/tests/test_fetch_coredumps.py

```diff
@@ -0,0 +1,139 @@
+import gzip
+import os
+
+class TestFetchCoreDumps(object):
+    class MockDecode(object):
+        def __init__(self, ret):
+            self.ret = ret
+            pass
+
+        def decode(self):
+            return self.ret
+
+    class MockPopen(object):
+        def __init__(self, ret):
+            self.ret = ret
+
+        def communicate(self, input=None):
+            return [TestFetchCoreDumps.MockDecode(self.ret)]
+
+    class MockRemote(object):
+        def __init__(self, system_type=None):
+            self.system_type = system_type
+
+        def sh(self, input=None):
+            return ""
+
+        def _sftp_get_file(self, path=None, input=None):
+            pass
+
+        def get_file(self, path=None, input=None):
+            pass
+
+    class MockPath(object):
+        def __init__(self, ret):
+            self.ret = ret
+
+        def join(self, path=None, input=None):
+            return self.ret
+
+        @property
+        def sep(self, path=None, input=None):
+            return self.ret
+
+        def isdir(self, input=None):
+            return True
+
+        def dirname(self, input=None):
+            return self.ret
+
+        def exists(self, input=None):
+            pass
+
+    def setup(self):
+        self.the_function = fetch_binaries_for_coredumps
+        with gzip.open('file.gz', 'wb') as f:
+            f.write(b'Hello world!')
+        self.core_dump_path = "file.gz"
+
+    def side_effect_func(self, compressed, correct):
+
+        uncompressed_correct = self.MockPopen(
+            "ELF 64-bit LSB core file,"\
+            " x86-64, version 1 (SYSV), SVR4-style, from '-bash' ,"\
+            " real uid: 1194, effective uid: 1194, real gid: 1194,"\
+            " effective gid: 1194, execfn: '/bin/bash', platform: x86_64"
+        )
+        uncompressed_incorrect = self.MockPopen("ASCII text")
+
+        compressed_correct = self.MockPopen(
+            "gzip compressed data, was "\
+            "'correct.format.core', last modified: Wed Jun 29"\
+            " 19:55:29 2022, from Unix, original size modulo 2^32 3167080"
+        )
+
+        compressed_incorrect = self.MockPopen(
+            "gzip compressed data, was "\
+            "'incorrect.format.core', last modified: Wed Jun 29"\
+            " 19:56:56 2022, from Unix, original size modulo 2^32 11"
+        )
+
+        if compressed:
+            if correct:
+                return [compressed_correct, uncompressed_correct]
+            else:
+                return [compressed_incorrect, uncompressed_incorrect]
+        else:
+            if correct:
+                return [uncompressed_correct, uncompressed_correct]
+            else: 
+                return [uncompressed_incorrect, uncompressed_incorrect]
+
+    # Core is not compressed and file is in the correct format
+    @patch('teuthology.task.internal.subprocess.Popen')
+    @patch('teuthology.task.tests.test_fetch_coredumps.TestFetchCoreDumps.MockRemote')
```

Comment:
I think I'm more confused now; won't `MockRemote` just become a `Mock` here?

Reply:
Yea there is definitely some cleaning up that I forgot to do. Thanks for catching it.